src/main/kotlin/com/cosmotech/api/utils/KubernetesService.kt

```diff
@@ -0,0 +1,119 @@
+    val api = checkKubernetesContext()
+    val secretNameLower = secretName.lowercase()
+    val labelSelector = buildLabelSector(secretNameLower)
+
+    try {
+      val secrets =
+          api.listNamespacedSecret(
+              namespace, null, null, null, null, labelSelector, null, null, null, null, null)
+      if (secrets.items.isEmpty()) {
+        logger.debug("Secret does not exists in namespace $namespace: cannot delete it")
+      } else {
+        logger.info("Secret exists in namespace $namespace: deleting it")
+        api.deleteNamespacedSecret(secretNameLower, namespace, null, null, null, null, null, null)
+      }
+    } catch (e: ApiException) {
+      logger.error("Kubernetes API Exception when listing or deleting secret ${e.message}")
+      throw e
+    }
+  }
+
+  private fun getSecretFromKubernetes(namespace: String, secretName: String): Map<String, String> {
+    val api = checkKubernetesContext()
+    val secretNameLower = secretName.lowercase()
+    val result = api.readNamespacedSecret(secretNameLower, namespace, "")
+
+    logger.debug("Secret retrieved for namespace $namespace")
+    return result.data?.mapValues { Base64.getDecoder().decode(it.value).toString(Charsets.UTF_8) }
+        ?: mapOf()
+  }
+
+  private fun createOrReplaceSecretIntoKubernetes(
+      namespace: String,
+      secretName: String,
+      secretData: Map<String, String>
+  ) {
+    val api = checkKubernetesContext()
+    logger.debug("Creating secret $secretName in namespace $namespace")
+
+    val secretNameLower = secretName.lowercase()
+    val labelSelector = buildLabelSector(secretNameLower)
+    var replace = false
+
+    try {
+      val secrets =
+          api.listNamespacedSecret(
+              namespace, null, null, null, null, labelSelector, null, null, null, null, null)
+      if (secrets.items.isEmpty()) {
+        logger.debug("Secret does not exists in namespace $namespace: creating it")
+      } else {
+        logger.debug("Secret already exists in namespace $namespace: replacing it")
+        replace = true
+      }
+    } catch (e: ApiException) {
+      logger.error("Kubernetes API Exception when listing secret ${e.message}")
+      throw e
+    }
+
+    val metadata = V1ObjectMeta()
+    metadata.name = secretNameLower
+    metadata.namespace = namespace
+    metadata.labels = mapOf(SECRET_LABEL to secretNameLower)
+    val body = V1Secret()
+    body.metadata = metadata
+
+    body.data = secretData.mapValues { Base64.getEncoder().encode(it.value.toByteArray()) }
+    body.type = "Opaque"
+
+    try {
+      if (replace) {
+        api.replaceNamespacedSecret(secretNameLower, namespace, body, null, null, null, null)
+      } else {
+        api.createNamespacedSecret(namespace, body, null, null, null, null)
+      }
+      logger.info("Secret created/replaced")
+    } catch (e: ApiException) {
+      logger.error("Kubernetes API Exception when creating/replacing secret ${e.message}")
+      throw e
+    }
```

Comment:
This can be put in a single try{}catch{} block

Reply:
IMHO no need at all these try catch.   Only @AterThrowing(APIException.class) advice for logging